canvas_modules/common-canvas/src/common-properties/controls/numberfield/numberfield.jsx

```diff
@@ -163,6 +163,7 @@ class NumberfieldControl extends React.Component {
 					allowEmpty
 					hideSteppers={this.props.tableControl || (this.props.control.controlType === ControlType.NUMBERFIELD)}
 					helperText={this.props.control.helperText}
+					readOnly={this.props.control.readOnly || this.props.readOnly}
```

Comment:
Same as above.  

Seems like at the level when creating the control we should just always pass down the readOnly prop.  Then not use `this.props.control.readOnly` inside the controls.